### DISCUSSION TITLE
Installing an extension to run in-browser (PPM Viewer)

### DISCUSSION THREAD

**Discussion by BenVella** [coder/code-server#3748] [Q&A] [answered]
I've pulled the PPM/PGM Viewer vsix from: https://www.vsixhub.com/vsix/57562/

Uploaded to remote and installed it.  It did appear under extensions but was left disabled (cannot enable).  The goal is to be able to view a generated ppm file from the vscode (in-browser).  The extension works fine locally but it doesn't appear to work in browser using vscode remote.

Vscode remote is deployed on railway.app as a docker container.  If I can get this PPM extension viewer to work and display the image for me it would be perfect!  Any suggestions on how to get an extension to run "locally" in browser, allowing me to interpret the ppm data file into a viewable image?

Thanks

**Comment by jsjoeio** [comment]
Could you tell me more about your setup? Do you mean "code-server" is deployed on railway?

**Comment by BenVella** [comment, marked as answer]
That is correct using code-server-deploy repo on this account. So it's as a
docker container.

It's used in combination with gcc10 and gdb to run some simple ray tracing
code and the result is a generated ppm image.

On the served browser vscode of vscode remote Id like to view this image as
I do locally using the mentioned extension and even though installing
through vsix has it show up in extensions it cannot be enabled and does not
work

**Reply by jsjoeio** [reply]
Thanks for clarifying! I have code-server set up on Railway so let me see if I can reproduce. 

Had trouble finding a ppm file online. I recorded a video trying to reproduce. Let me know if it helps us get any further on this. Feel free to share a link to a .ppm file

https://www.loom.com/share/ef3972506f7b4be19d0350c5cc9e5b03

**Reply by BenVella** [reply]
Thanks for going through this js, I retried the extension after it magically loaded for you on refresh and the ppm file did indeed load (afer a while).  I’m not sure why it wouldn’t be enabled no matter what I did previously and then it worked right away when I reinstalled it.

Regards,

Ben

**Reply by jsjoeio** [reply]
Strange! Well I'm glad it's working now.
